repos/system_upgrade/el7toel8/actors/scanmemory/libraries/library.py

```diff
@@ -0,0 +1,21 @@
+from leapp.libraries.stdlib import api
+from leapp.models import MemoryInfo
+
+
+def _get_memoryinfo(filename='/proc/meminfo'):
+    """ Returns dict of all memory information from /proc/meminfo file """
+
+    # Expecting the file exists on earch system
+    try:
+        with open(filename, 'r') as fp:
+            # format of the lines: Key:  value unit
+            # e.g.:                MemTotal:   1024 kB
+            return dict((i.split()[0].rstrip(':'), int(i.split()[1])) for i in fp.readlines())
+    except IOError:
+        return None
```

Comment:
This is still not resolving the original concerns we reported during previous reviews. Now in case it will fail, we will not see anything and will have to spend extra time with investigation what's the problem. If you have any questions, ping me or @fernflower to resolve them.